lib/gc/ExecutionEngine/OpenCLRuntime/OpenCLRuntimeWrappers.cpp

```diff
@@ -447,29 +528,35 @@ extern "C" OCL_RUNTIME_EXPORT void gpuWait(GPUCLQUEUE *queue) {
 
 // a silly workaround for mgpuModuleLoad. OCL needs context and device to load
 // the module. We remember the last call to any mgpu* APIs
-static thread_local GPUCLQUEUE *lastQueue;
+static thread_local GPUCLQUEUE *lastQueue{nullptr};
 extern "C" OCL_RUNTIME_EXPORT GPUCLQUEUE *mgpuStreamCreate() {
   auto ret =
       new GPUCLQUEUE(static_cast<cl_device_id>(nullptr), nullptr, nullptr);
   lastQueue = ret;
   return ret;
 }
 
+GPUCLQUEUE *getOrCreateStaticQueue() {
+  if (!lastQueue) {
+    return mgpuStreamCreate();
+  }
+  return lastQueue;
+}
+
 extern "C" OCL_RUNTIME_EXPORT void mgpuStreamDestroy(GPUCLQUEUE *queue) {
   lastQueue = nullptr;
   delete queue;
 }
 
 extern "C" OCL_RUNTIME_EXPORT void *
 mgpuMemAlloc(uint64_t size, GPUCLQUEUE *queue, bool isShared) {
-  lastQueue = queue;
-  return allocDeviceMemory(queue, size, /*alignment*/ 64, isShared);
+  return allocDeviceMemory(queue ? queue : getOrCreateStaticQueue(), size,
+                           /*alignment*/ 64, isShared);
 }
 
 extern "C" OCL_RUNTIME_EXPORT void mgpuMemFree(void *ptr, GPUCLQUEUE *queue) {
-  lastQueue = queue;
   if (ptr) {
-    deallocDeviceMemory(queue, ptr);
+    deallocDeviceMemory(queue ? queue : getOrCreateStaticQueue(), ptr);
```

Comment:
this looks a bit suspicious, what's the scenario for the case when we have to create a queue to deallocate memory?